src/lib/phdi-building-blocks/phdi_building_blocks/name_standardization.py

```diff
@@ -0,0 +1,59 @@
+from phdi_building_blocks.utils import find_patient_resources
+from typing import Callable
+
+
+def standardize_name(raw: str) -> str:
+    """trim spaces and force uppercase
+
+    >>> standardize_name(" JohN Doe ")
+    'JOHN DOE'
+    """
+
+    raw = [x for x in raw if not x.isnumeric()]
+    raw = "".join(raw)
+    raw = raw.upper()
+    raw = raw.strip()
+    return raw
+
+
+def standardize_patient_name(
+    bundle: dict, standardize: Callable = standardize_name
+) -> dict:
+    """Given a FHIR bundle and a standardization function, standardize the patient names
+    in all patient resources in the bundle. By default the standardize_name function
+    defined in this module is used, but the user is free to provide their own function
+    specifying a standardization process as long as it accepts and returns a string."""
+
+    for resource in find_patient_resources(bundle):
+        patient = resource.get("resource")
+        if "extension" not in patient:
+            patient["extension"] = []
+
+        # Transform names
+        for name in patient.get("name", []):
+            if "family" in name:
+                std_family = standardize(name["family"])
+                raw_family = name["family"]
+                patient["extension"].append(
+                    {
+                        "url": "http://usds.gov/fhir/phdi/StructureDefinition/family-name-was-standardized",  # noqa
+                        "valueBoolean": raw_family != std_family,
+                    }
+                )
```

Comment:
Something we'll have to consider when we start importing using the FHIR server is that the extensions we're appending for metrics tracking don't actually resolve to valid URLs. Perhaps not a problem, but if a user's data suddenly shows up with some "garbage" / "fake" extensions in there that we used for our purposes that might be seen as losing data quality or a reduction in data cleanliness.

Reply:
IMO there isn't anything wrong with us defining our own extensions, and I can see information about what data was standardized being valuable to STLTs even beyond metrics collection.  I think it would be good to discuss/reach consensus about if others agree.  We could also discuss if this is the right level to record this info, or whether we wanted, for example, to collect the original value received.  

Here's an example of a vendor (one I worked a lot with in a past life) who publishes a list of extensions.  
https://docs.intersystems.com/components/csp/docbook/fhir_extn_sda3/index.html

Navigating to any of the URLs listed in this page will redirect to the page itself.  If we want to define our own extensions (which I assume we will for one reason or another), we would just need to publish them similarly.

Reply:
That makes sense, Spencer. I defer to your expertise in this domain because you have a _lot_ more familiarity with how extensions are used in the FHIR server in practice, whereas the only things I know are that the FHIR pages tell me certain things in bold letters 😂 . So if it seems okay to you I'm happy to keep them as is, as well.

Reply:
Sounds good!  My hope is that we will all learn a lot about how the FHIR server operates as we move into the future.